### PR TITLE
Redux in react

### PR DESCRIPTION
**Pull request by AlyAtalla**
Create an initialState variable for your booksSlice, which will includes:
// Initial state:
[
  {
    "item_id": "item1",
    "title": "The Great Gatsby",
    "author": "John Smith",
    "category": "Fiction"
  },
  {
    "item_id": "item2",
    "title": "Anna Karenina",
    "author": "Leo Tolstoy",
    "category": "Fiction"
  },
  {
    "item_id": "item3",
    "title": "The Selfish Gene",
    "author": "Richard Dawkins",
    "category": "Nonfiction"
  }
]
Display books, received from the slice, in a reusable component
Dispatch actions using useDispatch
Add a <Button> component, which includes:
An event handler that adds a book to the books array 
Add a <Button> component, which includes:
An event handler that removes a book from the books array

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,19 +1,35 @@
...
+          <div className="content-container d-flex flex-column justify-center align-center">
+            <Navbar />
+            <Routes>
+              <Route path="/categories" element={<Categories />} />
+              <Route exact path="/" element={<BooksPage />} />
+            </Routes>
+          </div>
+        </div>
+      </Router>
+    </Provider>
```
Comment (V-Blaze) [inline src/App.js]: - You have done great so far, :+1: :+1: :100: but there are few changes you need to make before moving to the next milestone. :+1: :100: 

- [ ] You did great wrapping the React-App with the `<Provider>` component from `react-redux`. But I'll suggest you move this to the `index.js` file.as that's the best convention. :+1:

File: src/App.js [hunk 2]
```diff
@@ -1,19 +1,35 @@
-import './App.css';
+/* eslint-disable import/no-unresolved */
+/* eslint-disable import/extensions */
+/* eslint-disable no-unused-vars */
```
Comment (V-Blaze) [inline src/App.js]: - [ ] As required by MIcroverse, you are not meant to use any LInters diable rule in your codebase. The Linters are there to make sure you follow the best practices. I'll suggest you remove these and try to fix the errors. :smile: :small_airplane: :+1:

File: src/components/Book.jsx [hunk 3]
```diff
@@ -0,0 +1,70 @@
+/* eslint-disable */
```
Comment (V-Blaze) [inline src/components/Book.jsx]: - [ ] Just as I pointed out above, I'll suggest you remove these linters disable rule and try to fix the errors. :+1: :small_airplane: :smile:

File: src/components/Book.jsx [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+import BookstoreAPI from '../services/BookstoreAPI';
+
+const Book = ({
+  id, category, title, author, currentChapter, totalChapters,
+}) => {
+  const dispatch = useDispatch();
+  const handleClick = () => {
+    dispatch(BookstoreAPI.deleteBookFromAPI(id));
+    dispatch(BookstoreAPI.getAllBooksFromAPI());
+  };
```
Comment (V-Blaze) [inline src/components/Book.jsx:13]: - [ ] As stated in the project requirement, you are meant to be adding and removing books from the local redux store and not the API, I'll suggest you update your code to dispatch actions `addBook, removeBook` you created in the redux store. :+1: :+1:

File: src/components/BookForm.jsx [hunk 5]
```diff
@@ -0,0 +1,90 @@
+/* eslint-disable */
+/* eslint-disable import/no-duplicates */
```
Comment (V-Blaze) [inline src/components/BookForm.jsx]: - [ ] Just as I pointed out above, I'll suggest you remove these linters disable rule and try to fix the errors. :+1: :small_airplane: :smile:

File: src/components/BookForm.jsx [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+          ADD BOOK
+        </button>
+
+      </form>
+    </div>
+  );
+};
+
+export default BookForm;
+/* eslint-enable */
```
Comment (V-Blaze) [inline src/components/BookForm.jsx]: - [ ] Just as I pointed out above, I'll suggest you remove these linters disable rule and try to fix the errors. :+1: :small_airplane: :smile:

File: src/components/BookList.jsx [hunk 7]
```diff
@@ -0,0 +1,40 @@
+/* eslint-disable */
+/* eslint-disable react-hooks/exhaustive-deps */
```
Comment (V-Blaze) [inline src/components/BookList.jsx]: - [ ] Just as I pointed out above, I'll suggest you remove these linters disable rule and try to fix the errors. :+1: :small_airplane: :smile:  also remove the one in the last line of this file. :+1:

File: src/components/Navbar.jsx [hunk 8]
```diff
@@ -0,0 +1,37 @@
+/* eslint-disable */
```
Comment (V-Blaze) [inline src/components/Navbar.jsx]: - [ ] Just as I pointed out above, I'll suggest you remove these linters disable rule and try to fix the errors. :+1: :small_airplane: :smile:  also remove the one in the last line of this file. :+1:

File: src/containers/BooksPage.jsx [hunk 9]
```diff
@@ -0,0 +1,14 @@
+/* eslint-disable */
```
Comment (V-Blaze) [inline src/containers/BooksPage.jsx]: - [ ] Just as I pointed out above, I'll suggest you remove these linters disable rule and try to fix the errors. :+1: :small_airplane: :smile:  also remove the one in the last line of this file. :+1:

File: src/containers/Categories.jsx [hunk 10]
```diff
@@ -0,0 +1,11 @@
+/* eslint-disable */
```
Comment (V-Blaze) [inline src/containers/Categories.jsx]: - [ ] Just as I pointed out above, I'll suggest you remove these linters disable rule and try to fix the errors. :+1: :small_airplane: :smile:  also remove the one in the last line of this file. :+1:

File: src/index.js [hunk 11]
```diff
@@ -1,18 +1,12 @@
+/* eslint-disable no-unused-vars */
```
Comment (V-Blaze) [inline src/index.js]: - [ ] Just as I pointed out above, I'll suggest you remove these linters disable rule and try to fix the errors. :+1: :small_airplane: :smile:  also remove the one in the last line of this file. :+1:

File: src/redux/books/books.js [hunk 12]
```diff
@@ -0,0 +1,56 @@
...
+      return { books: action.payload };
+    case ADD_BOOK:
+      return { books: [...state.books, action.payload] };
+    case REMOVE_BOOK:
+      return { books: state.books.filter((book) => book.id !== action.payload) };
+    default:
+      return state;
+  }
+};
+/* eslint-enable */
```
Comment (V-Blaze) [inline src/redux/books/books.js]: - [ ] Just as I pointed out above, I'll suggest you remove these linters disable rule and try to fix the errors. :+1: :small_airplane: :smile:  
- [ ]

File: src/services/BookstoreAPI.js [hunk 13]
```diff
@@ -0,0 +1,51 @@
+/* eslint-disable no-undef */
```
Comment (V-Blaze) [inline src/services/BookstoreAPI.js:1]: - [ ] Just as I pointed out above, I'll suggest you remove these linters disable rule and try to fix the errors. :+1: :small_airplane: :smile:  also remove the one in the last line of this file. :+1:

File: src/redux/books/books.js [hunk 14]
```diff
@@ -0,0 +1,56 @@
...
+      author: 'Leo Tolstoy',
+      category: 'Fiction',
+    },
+    {
+      item_id: 'item3',
+      title: 'The Selfish Gene',
+      author: 'Richard Dawkins',
+      category: 'Nonfiction',
+    },
+  ],
```
Comment (V-Blaze) [inline src/redux/books/books.js:25]: - [ ] When running Ypour app, looking at the UI you can see that the books shown in the UI are not the book in this initial array of books in the redux state. I'll suggest you make use of `useSelector ` and these books out from the reudux store and display them in the UI. :+1: 

- - Screenshot:
Screenshot from 2023-08-21 11-37-49

**Review by V-Blaze — CHANGES_REQUESTED** [review]
## Hi @AlyAtalla 

## Required changes 🖊️ 
# Almost there 🟡 🟦 🟨 

almost-there


Great work so far, You have come a long way, keep moving forward👏 👍  
You've done well implementing the project requirements, but there are still some issues I think you need to improve on. 😄 👋 


 ## Highlights
- Correct Gitflow✔
- Readme looks professinal👋 
-  Used correct linters for project✔

 ## Required Changes ♻️

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/redux/books/books.js (hunk 14 above)
Comment (AlyAtalla) [reply src/redux/books/books.js:25]: thank you

File: src/App.js [hunk 15]
```diff
@@ -1,19 +1,32 @@
...
+          <div className="content-container d-flex flex-column justify-center align-center">
+            <Navbar />
+            <Routes>
+              <Route path="/categories" element={<Categories />} />
+              <Route exact path="/" element={<BooksPage />} />
+            </Routes>
+          </div>
+        </div>
+      </Router>
+    </Provider>
```
Comment (skyv26) [inline src/App.js]: - [x] Good job by implementing the changes asked by your previous code reviewer but It seems like you made the changes but maybe forget to shift your `Provider` store code from `App.js` to `index.js`. Well your previous code reviewer mentioned that it is a better convention and help you to maintain the code easily.

**Review by skyv26 — CHANGES_REQUESTED** [review]
## Hi @AlyAtalla, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

## STATUS: REQUIRE CHANGES 🔴 



## Highlights
- Good **PR** title ✅ 
- Descriptive *PR* description ✅ 
- Implement the features ✅ 
- Nice *UI* design ✅ 
- Professional `README` template ✅ 
- Linters are green ✅ 
- Correct `GitFlow` configuration ✅ 
- Meaningful **Github** commits ✅ 
- Correct `.github/workflows` for linters ✅ 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

#### Cheers, and Happy coding!👏👏👏

## ⚠️  WARNING ⚠️ 
**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

## Have any doubt ❓ 
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag @skyv26 in your question so I can receive the notification. You can also follow me on the below platforms

<table>
<tr>
<td>

<a href="https://github.com/skyv26">
   @skyv26
</a>

</td>
<td>

<a href="https://www.linkedin.com/in/skyv2022/">
   linkedin-skyv2022
</a>

</td>
<td>

<a href="https://twitter.com/vrma_aakash">
   @vrma_aakash
</a>

</td>
<td>

<a href="https://www.instagram.com/skybrel/">
   @skybrel
</a>

</td>
<td>

<a href="https://www.frontendmentor.io/profile/skyv26">
   @skybrel
</a>

</td>
</tr>
</table>

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/App.js (hunk 15 above)
Comment (AlyAtalla) [reply src/App.js]: thank you

**Review by AbelG101 — APPROVED** [review]
# Hi @AlyAtalla 👋, 

**Your project is complete!  There is nothing else to say other than... it's time to merge it. 
Congratulations! 🎉 🥳🥇**

## Highlights
- Proper use of `<Provider>` component ✔
- Proper use of `useSelector()` ✔
- Books can be properly added and removed ✔
- Redux state is properly managed ✔
- Correct use of Git Flow ✔
-  Well-structured files ✔
- All linter checks are passing ✔

## Optional suggestions
- N/A

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**